metalarchives_importer.user.js

```diff
@@ -0,0 +1,242 @@
+    var patt = new RegExp('^d{4}$');
+    if (patt.exec(datestring)) {
+        release.year = datestring;
+    } else if (datestring.indexOf(',') != - 1) {
+        var commaindex = datestring.indexOf(',');
+        var d = new Date(datestring.substring(0, commaindex - 2) + datestring.substring(commaindex));
+        release.year = d.getFullYear();
+        release.month = d.getMonth()+1;
+        release.day = d.getDate();
+        console.log(release.day);
+    } else{
+        var d = new Date("2 "+datestring);
+        release.year = d.getFullYear();
+        release.month = d.getMonth()+1;
+        console.log(release.day);
+    }
+    return release;
+}
+
+function getGenericalData(){
+    var rdata=new Array();
+    var keydata = $('dl.float_left dt, dl.float_right dt').map(function(){
+        var s=$.trim($(this).text());
+        return s.substring(0,s.length-1);
+    } ).get();
+    var valuedata= $('dl.float_left dd,dl.float_right dd').map(function(){
+        return $.trim($(this).text());
+    } ).get();
+    for(i=0;i<keydata.length;i++){
+        rdata[keydata[i]]=valuedata[i];
+    }
+    console.log(rdata);
+    return rdata;
+}
+function retrieveReleaseInfo() {
+    var release = new Object();
+
+    release.artist_credit =new Array();
+    release.artist_credit.push({artist_name: $('h2.band_name').text(),credited_name: $('h2.band_name').text()});
+    release.title = $('h1.album_name').text();
+
+    var rdata=getGenericalData();
+
+    release = setreleasedate(release, rdata["Release date"]);
+    //todo add case for multiple labels if such a case exist
+    release.labels = new Array();
+    release.labels.push( { 
+        name: (rdata["Label"]== "Independent"?"[no label]":rdata["Label"]), 
+        catno: (rdata["Catalog ID"] == "N/A" ? "" :rdata["Catalog ID"]) 
+    } );
+
+    release.country = ""; // Worldwide
+
+
+    release.type =ReleaseTypes[rdata["Type"]][0];
+    release.secondary_types = new Array(ReleaseTypes[rdata["Type"]][1]);
+    release.status = 'official';
+
+    rdata.hasOwnProperty("Version desc.") &&  rdata["Version desc."].indexOf("Digipak")!=-1?release.packaging ="Digipak":release.packaging ="";
+    release.language = '';
+    release.script = '';
+    release.barcode='';
+    var identifiers=$("#album_tabs_notes > div:nth-child(2)").find("p:not([class])").contents();
+    for(var j=0;j<identifiers.length;j++){
+        if(identifiers[j].textContent.indexOf("Barcode:")!=-1){
+            release.barcode=$.trim(identifiers[j].textContent.substring(8));
+            break;
+        }
+    }
+    // Release URL
+    release.urls = new Array();
+    release.urls.push( { url: window.location.href, link_type: 82 } );
+
+    // Tracks
+    /*discs = [
+            {
+                title,
+                format,
+                tracks = [
+                    { number, title, duration, artist_credit },
+                    ...
+                ]*/
+    var releaseNumber=1;
```

Comment:
I can check it here: http://www.metal-archives.com/albums/Reverend_Bizarre/Death_Is_Glory..._Now/219935